src/symbol/sprite_atlas.js

```diff
@@ -50,6 +54,68 @@ class SpriteAtlas {
         return rect;
     }
 
+    addImage(name, pixels, options) {
+        let width, height, pixelRatio;
+        if (pixels instanceof window.HTMLImageElement) {
+            width = pixels.width;
+            height = pixels.height;
+            pixels = browser.getImageData(pixels);
+            pixelRatio = this.pixelRatio;
+        } else {
+            width = options.width;
+            height = options.height;
+            pixelRatio = options.pixelRatio || this.pixelRatio;
+        }
+
+        if (ArrayBuffer.isView(pixels)) {
+            pixels = new Uint32Array(pixels.buffer);
+        }
+
+        if (!(pixels instanceof Uint32Array)) {
+            this.fire('error', {error: new Error('Image provided in an invalid format. Supported formats are HTMLImageElement, ImageData, and ArrayBufferView.')});
+        }
+
+        if (this.images[name]) {
+            this.fire('error', {error: new Error('An image with this name already exists.')});
+        }
+
+        const rect = this.allocateImage(width, height);
+        if (!rect) {
+            this.fire('error', {error: new Error('There is not enough space to add this image.')});
+        }
+
+        const image = new AtlasImage(rect, width / pixelRatio, height / pixelRatio, false, 1);
+        this.images[name] = image;
+
+        this.copy(pixels, width, rect, {pixelRatio, x: 0, y: 0, width, height}, false);
+
+        this.fire('data', {dataType: 'style'});
+    }
+
+    removeImage(name) {
+        const image = this.images[name];
+        delete this.images[name];
+
+        if (!image) {
+            this.fire('error', {error: new Error('No image with this name exists.')});
+        }
+
+        const rect = image.rect;
+        const pixelRatio = image.pixelRatio;
+
+        this.copy(
+            new Uint32Array(image.width * image.height * pixelRatio * pixelRatio),
+            image.width * pixelRatio,
+            rect,
+            {pixelRatio: pixelRatio, x: 0, y: 0, width: image.width * pixelRatio, height: image.width * pixelRatio},
+            false
+        );
```

Comment:
I added this code out of fear of edge cases with semitransparent images & linear interpolation. Testing suggests these concerns are unwarranted. Removing it for now.